### ISSUE TITLE
Geth API: incorrect parsing of debug_traceBlockByNumber RPC response

### ISSUE THREAD

**Issue by frostiq** [Nethereum/Nethereum#706]
## Current Behavior
The following exception occurs when parsing debug_traceBlockByNumber response:
```
Unable to convert the result to type Newtonsoft.Json.Linq.JObject

   at Nethereum.JsonRpc.Client.RpcMessages.RpcResponseExtensions.GetResult[T](RpcResponseMessage response, Boolean returnDefaultIfNull, JsonSerializerSettings settings)
   at Nethereum.JsonRpc.Client.ClientBase.SendInnerRequestAsync[T](RpcRequestMessage reqMsg, String route)
   at Nethereum.JsonRpc.Client.ClientBase.SendInnerRequestAsync[T](RpcRequest request, String route)
   at Nethereum.JsonRpc.Client.ClientBase.SendRequestAsync[T](RpcRequest request, String route)
   at Bloqboard.Conto.TxProcessor.Infrastructure.MaticDepositsBlockScanner.ScanBlock(HexBigInteger blockNumber) in 
```
## Steps to Reproduce
```
var response = await _debugApiService.TraceBlockByNumber.SendRequestAsync(blockNumber.ToUlong());
```

## Context (Environment)
bor/v1.10.8-stable-045e5c2c/linux-amd64/go1.15.5

## Possible Implementation
```
public interface IDebugTraceBlockByNumber
{
    RpcRequest BuildRequest(ulong blockNumber, object id = null);

    Task<JArray> SendRequestAsync(ulong blockNumber, object id = null);
}

public class DebugTraceBlockByNumber : RpcRequestResponseHandler<JArray>, IDebugTraceBlockByNumber
{
    public DebugTraceBlockByNumber(IClient client)
        : base(client, ApiMethods.debug_traceBlockByNumber.ToString())
    {
    }

    public RpcRequest BuildRequest(ulong blockNumber, object id = null) => this.BuildRequest(id, new HexBigInteger(blockNumber));

    public Task<JArray> SendRequestAsync(ulong blockNumber, object id = null) => this.SendRequestAsync(id, new HexBigInteger(blockNumber));
}
```

**Comment by frostiq** [comment]
Not sure though if the issue holds for the original Geth or just for Bor/Polygon fork.
If anyone has a Geth node up and running with archive mode turned on, please double check if the issue persists with the Ethereum network

**Comment by frostiq** [comment]
Addressed in #707
